numpy/linalg/linalg.py

```diff
@@ -1608,24 +1608,29 @@ def svd(a, full_matrices=True, compute_uv=True, hermitian=False):
     True
 
     """
+    import numpy as _nx
     a, wrap = _makearray(a)
 
     if hermitian:
-        # note: lapack returns eigenvalues in reverse order to our contract.
-        # reversing is cheap by design in numpy, so we do so to be consistent
+        # note: lapack returns eigenvalues with s ** 2 sorted instead of s.
+        # so we re-order the eigenvalues and related arrays to have the
+        # correct order
         if compute_uv:
             s, u = eigh(a)
-            s = s[..., ::-1]
-            u = u[..., ::-1]
-            # singular values are unsigned, move the sign into v
-            vt = transpose(u * sign(s)[..., None, :]).conjugate()
+            sgn = sign(s)
             s = abs(s)
+            sidx = argsort(s)[..., ::-1]
+            sgn = _nx.take_along_axis(sgn, sidx, axis=-1)
+            s = _nx.take_along_axis(s, sidx, axis=-1)
+            u = _nx.take_along_axis(u, sidx[..., None, :], axis=-1)
```

Comment:
I'm not convinced whether this is the right way around (`None, :` vs `:, None`). 

Do tests fail if you swap them?

Reply:
> Do tests fail if you swap them?

I just checked, and they do. Reason about it this way: The last dimension (columns) is the one that has to be sorted, so we have to place the last axis of `sidx` parallel to it.

This is matching with the `None, :` below. If you like I could add a comment.